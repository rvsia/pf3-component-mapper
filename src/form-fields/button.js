import React from 'react';
import { Button } from 'patternfly-react';

export default (props) => <Button style={{ marginLeft: 3 }} { ...props } />;
