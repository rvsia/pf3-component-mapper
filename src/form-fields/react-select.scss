$input-default-box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);

.final-form-select.has-error{
  border: 1px solid;
  border-color: #cc0000;
}

.form-group.field-group:not(:last-child) {
  margin-bottom: 0;
}
.form-group.field-group:last-child {
  margin-bottom: 0;
}

div.field-group {
  margin-bottom: 25px;
}