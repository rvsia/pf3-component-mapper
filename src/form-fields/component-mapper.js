import React from 'react';
import { componentTypes } from '@data-driven-forms/react-form-renderer';
import { TextField, TextareaField, SelectField, CheckboxGroup, Radio } from './form-fields';
import SubForm from './sub-form';
import Tabs from './tabs';

const mapper = componentType => ({
  [componentTypes.TEXT_FIELD]: TextField,
  [componentTypes.TEXTAREA_FIELD]: TextareaField,
  [componentTypes.FIELD_ARRAY]: props => <div>field array</div>,
  [componentTypes.SELECT_COMPONENT]: SelectField,
  [componentTypes.FIXED_LIST]: props => <div>fixed list</div>,
  [componentTypes.CHECKBOX]: CheckboxGroup,
  [componentTypes.SUB_FORM]: SubForm,
  [componentTypes.RADIO]: Radio,
  [componentTypes.TABS]: Tabs,
  [componentTypes.TAB_ITEM]: props => <div>tab item</div>,
  [componentTypes.DATE_PICKER]: props => <div>date picker</div>,
  [componentTypes.TIME_PICKER]: props => <div>time picker</div>,
  [componentTypes.TAG_CONTROL]: props => <div>tag control</div>,
})[componentType];

export default mapper;
