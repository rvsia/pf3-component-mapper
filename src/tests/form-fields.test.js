import React from 'react';
import toJson from 'enzyme-to-json';
import { SwitchField, CheckboxGroup, SelectField, Radio, TextareaField, TextField } from '../form-fields/form-fields';
import { mount } from 'enzyme';
import MockFieldProvider from './mock-field-provider';
import MultipleChoiceList from '../form-fields/multiple-choice-list';
import Select from 'react-select';

describe('FormFields', () => {
  describe('<SwitchField />', () => {
    const props = {
      input: {
        name: 'Name of the field',
        value: '',
      },
      id: 'someIdKey',
      dataType: 'someDataType',
      meta: {
        error: false,
        touched: false,
      },
      FieldProvider: MockFieldProvider,
    };

    it('should render Switch correctly', () => {
      const wrapper = mount(
        <SwitchField { ...props } />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render mini Switch correctly', () => {
      const wrapper = mount(
        <SwitchField { ...props } bsSize='mini' />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render sm Switch correctly', () => {
      const wrapper = mount(
        <SwitchField { ...props } bsSize='mn' />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render Switch with label correctly', () => {
      const wrapper = mount(
        <SwitchField { ...props } label={ 'Label' } />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render Switch with placeholder correctly', () => {
      const wrapper = mount(
        <SwitchField { ...props } placeholder={ 'Placeholder' } />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render Switch with onText (custom prop) correctly', () => {
      const wrapper = mount(
        <SwitchField { ...props } onText={ 'OnText' } />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render disabled Switch correctly', () => {
      const wrapper = mount(
        <SwitchField { ...props } isDisabled />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render readOnly Switch correctly', () => {
      const wrapper = mount(
        <SwitchField { ...props } isReadOnly />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should call onChange method', () => {
      const changeSpy = jest.fn();
      const wrapper = mount(
        <SwitchField
          { ...props }
          name="Foo"
          FieldProvider={ props => (
            <MockFieldProvider
              { ...props }
              input={{ onChange: changeSpy, name: props.name }}
            />
          ) }
        />
      );
      wrapper.find('input').simulate('change');
      expect(changeSpy).toHaveBeenCalled();
    });
  });

  describe('<CheckboxGroup />', () => {
    let initialProps;
    beforeEach(() => {
      initialProps = {
        FieldProvider: MockFieldProvider,
        input: {
          name: 'single-check-box',
        },
        meta: {},
      };
    });

    it('should render single checkbox variant', () => {
      const wrapper = mount(<CheckboxGroup { ...initialProps } />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render multiple choice variant', () => {
      const wrapper = mount(<CheckboxGroup { ...initialProps } options={ [] } />);
      expect(wrapper.find(MultipleChoiceList)).toHaveLength(1);
    });
  });

  describe('<SelectField />', () => {
    let initialProps;
    const changeSpy = jest.fn();
    beforeEach(() => {
      initialProps = {
        input: {
          name: 'select-input',
          onChange: changeSpy,
        },
        meta: {},
        options: [{
          label: 'option 1',
          value: 1,
        }, {
          label: 'option 2',
          value: 2,
        }],
      };
    });

    afterEach(() => {
      changeSpy.mockReset();
    });

    it('should mount correctly', () => {
      const wrapper = mount(<SelectField { ...initialProps } />);
      expect(wrapper).toBeTruthy();
    });

    it('should call on change with correct value on single select', () => {
      const wrapper = mount(<SelectField { ...initialProps } />);
      wrapper.find(Select).instance().props.onChange({
        value: 2,
      });
      expect(changeSpy).toHaveBeenCalledWith(2);
    });

    it('should call on change with correct value on multi select', () => {
      const wrapper = mount(
        <SelectField
          { ...initialProps }
          multi
          input= {{
            name: 'select-input',
            onChange: changeSpy,
            value: [],
          }}
        />);
      wrapper.find(Select).instance().props.onChange(
        [{ value: 2, label: 'x' }, { value: 1, label: 'a' }],
      );
      expect(changeSpy).toHaveBeenCalledWith([ 1, 2 ]);
    });
  });

  describe('<Radio />', () => {
    let initialProps;
    beforeEach(() => {
      initialProps = {
        input: {
          name: 'radio',
        },
        meta: {},
        options: [{
          label: 'option 1',
          value: 1,
        }, {
          label: 'option 2',
          value: 2,
        }],
        FieldProvider: MockFieldProvider,
      };
    });

    it('should render correctly', () => {
      const wrapper = mount(<Radio { ...initialProps } />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('<TextField />', () => {
    let initialProps;
    beforeEach(() => {
      initialProps = {
        input: {
          name: 'text-field',
        },
        meta: {},
      };
    });

    it('should render correctly', () => {
      const wrapper = mount(<TextField { ...initialProps } />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render correctly with placeholder', () => {
      const wrapper = mount(<TextField { ...initialProps } placeholder={ 'placeholder' } />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('<TextareaField />', () => {
    let initialProps;
    beforeEach(() => {
      initialProps = {
        input: {
          name: 'textarea-field',
        },
        meta: {},
      };
    });

    it('should render correctly', () => {
      const wrapper = mount(<TextareaField { ...initialProps } />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render correctly with placeholder', () => {
      const wrapper = mount(<TextareaField { ...initialProps } placeholder={ 'placeholder' } />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
