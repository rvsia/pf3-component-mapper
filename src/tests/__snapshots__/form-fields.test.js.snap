// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFields <CheckboxGroup /> should render single checkbox variant 1`] = `
<CheckboxGroup
  FieldProvider={[Function]}
  input={
    Object {
      "name": "single-check-box",
    }
  }
  meta={Object {}}
>
  <FieldInterface
    FieldProvider={[Function]}
    componentType="checkbox"
    input={
      Object {
        "name": "single-check-box",
      }
    }
    meta={Object {}}
    name="single-check-box"
  >
    <CheckboxGroupField
      FieldProvider={[Function]}
      componentType="checkbox"
      id="single-check-box"
      input={
        Object {
          "name": "single-check-box",
        }
      }
      meta={Object {}}
      name="single-check-box"
    >
      <FinalFormField
        FieldProvider={[Function]}
        componentType="checkbox"
        id="single-check-box"
        input={
          Object {
            "name": "single-check-box",
          }
        }
        meta={Object {}}
        name="single-check-box"
        noCheckboxLabel={true}
      >
        <FormGroup
          bsClass="form-group"
          validationState={null}
        >
          <div
            className="form-group"
          >
            <Col
              bsClass="col"
              componentClass="div"
              md={12}
            >
              <div
                className="col-md-12"
              >
                <Checkbox
                  bsClass="checkbox"
                  disabled={false}
                  inline={false}
                  name="single-check-box"
                  title=""
                >
                  <div
                    className="checkbox"
                  >
                    <label
                      title=""
                    >
                      <input
                        disabled={false}
                        name="single-check-box"
                        type="checkbox"
                      />
                    </label>
                  </div>
                </Checkbox>
              </div>
            </Col>
          </div>
        </FormGroup>
      </FinalFormField>
    </CheckboxGroupField>
  </FieldInterface>
</CheckboxGroup>
`;

exports[`FormFields <Radio /> should render correctly 1`] = `
<Radio
  FieldProvider={[Function]}
  input={
    Object {
      "name": "radio",
    }
  }
  meta={Object {}}
  options={
    Array [
      Object {
        "label": "option 1",
        "value": 1,
      },
      Object {
        "label": "option 2",
        "value": 2,
      },
    ]
  }
>
  <FieldInterface
    FieldProvider={[Function]}
    componentType="radio"
    input={
      Object {
        "name": "radio",
      }
    }
    meta={Object {}}
    name="radio"
    options={
      Array [
        Object {
          "label": "option 1",
          "value": 1,
        },
        Object {
          "label": "option 2",
          "value": 2,
        },
      ]
    }
  >
    <FormGroup
      bsClass="form-group"
      validationState={null}
    >
      <div
        className="form-group"
      >
        <Col
          bsClass="col"
          componentClass="div"
          md={12}
        >
          <div
            className="col-md-12"
          >
            <MockFieldProvider
              key="radio-1"
              name="radio"
              render={[Function]}
              type="radio"
              value={1}
            >
              <Radio
                bsClass="radio"
                disabled={false}
                inline={false}
                onChange={[Function]}
                title=""
              >
                <div
                  className="radio"
                >
                  <label
                    title=""
                  >
                    <input
                      disabled={false}
                      onChange={[Function]}
                      type="radio"
                    />
                    option 1
                  </label>
                </div>
              </Radio>
            </MockFieldProvider>
            <MockFieldProvider
              key="radio-2"
              name="radio"
              render={[Function]}
              type="radio"
              value={2}
            >
              <Radio
                bsClass="radio"
                disabled={false}
                inline={false}
                onChange={[Function]}
                title=""
              >
                <div
                  className="radio"
                >
                  <label
                    title=""
                  >
                    <input
                      disabled={false}
                      onChange={[Function]}
                      type="radio"
                    />
                    option 2
                  </label>
                </div>
              </Radio>
            </MockFieldProvider>
          </div>
        </Col>
      </div>
    </FormGroup>
  </FieldInterface>
</Radio>
`;

exports[`FormFields <SwitchField /> should render Switch correctly 1`] = `
<SwitchField
  FieldProvider={[Function]}
  dataType="someDataType"
  id="someIdKey"
  input={
    Object {
      "name": "Name of the field",
      "value": "",
    }
  }
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
>
  <MockFieldProvider
    dataType="someDataType"
    id="someIdKey"
    input={
      Object {
        "name": "Name of the field",
        "value": "",
      }
    }
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    render={[Function]}
  >
    <FieldInterface
      componentType="switch-field"
      dataType="someDataType"
      id="someIdKey"
      input={
        Object {
          "name": "Name of the field",
          "onChange": [MockFunction],
          "value": "",
        }
      }
      meta={
        Object {
          "error": false,
          "touched": false,
        }
      }
      name="Name of the field"
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <Col
            bsClass="col"
            componentClass="div"
            md={12}
          >
            <div
              className="col-md-12"
            >
              <Switch
                checked=""
                id="someIdKey"
                name="Name of the field"
                offText="OFF"
                onChange={[Function]}
                onText="ON"
                value=""
              >
                <label
                  className="pf3-switch"
                  style={
                    Object {
                      "width": 80,
                    }
                  }
                >
                  <input
                    checked=""
                    id="someIdKey"
                    name="Name of the field"
                    onChange={[Function]}
                    type="checkbox"
                    value=""
                  />
                  <span
                    className="pf3-switch-slider"
                  >
                    <span
                      className="on-text"
                      style={
                        Object {
                          "left": -46,
                          "width": 46,
                        }
                      }
                    >
                      ON
                    </span>
                    <span
                      className="divider"
                      style={Object {}}
                    />
                    <span
                      className="off-text"
                      style={
                        Object {
                          "left": 34,
                          "width": 46,
                        }
                      }
                    >
                      OFF
                    </span>
                  </span>
                </label>
              </Switch>
            </div>
          </Col>
        </div>
      </FormGroup>
    </FieldInterface>
  </MockFieldProvider>
</SwitchField>
`;

exports[`FormFields <SwitchField /> should render Switch with label correctly 1`] = `
<SwitchField
  FieldProvider={[Function]}
  dataType="someDataType"
  id="someIdKey"
  input={
    Object {
      "name": "Name of the field",
      "value": "",
    }
  }
  label="Label"
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
>
  <MockFieldProvider
    dataType="someDataType"
    id="someIdKey"
    input={
      Object {
        "name": "Name of the field",
        "value": "",
      }
    }
    label="Label"
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    render={[Function]}
  >
    <FieldInterface
      componentType="switch-field"
      dataType="someDataType"
      id="someIdKey"
      input={
        Object {
          "name": "Name of the field",
          "onChange": [MockFunction],
          "value": "",
        }
      }
      label="Label"
      meta={
        Object {
          "error": false,
          "touched": false,
        }
      }
      name="Name of the field"
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <Col
            bsClass="col"
            className="control-label"
            componentClass="label"
            md={2}
          >
            <label
              className="control-label col-md-2"
            >
              Label
            </label>
          </Col>
          <Col
            bsClass="col"
            componentClass="div"
            md={10}
          >
            <div
              className="col-md-10"
            >
              <Switch
                checked=""
                id="someIdKey"
                name="Name of the field"
                offText="OFF"
                onChange={[Function]}
                onText="ON"
                value=""
              >
                <label
                  className="pf3-switch"
                  style={
                    Object {
                      "width": 80,
                    }
                  }
                >
                  <input
                    checked=""
                    id="someIdKey"
                    name="Name of the field"
                    onChange={[Function]}
                    type="checkbox"
                    value=""
                  />
                  <span
                    className="pf3-switch-slider"
                  >
                    <span
                      className="on-text"
                      style={
                        Object {
                          "left": -46,
                          "width": 46,
                        }
                      }
                    >
                      ON
                    </span>
                    <span
                      className="divider"
                      style={Object {}}
                    />
                    <span
                      className="off-text"
                      style={
                        Object {
                          "left": 34,
                          "width": 46,
                        }
                      }
                    >
                      OFF
                    </span>
                  </span>
                </label>
              </Switch>
            </div>
          </Col>
        </div>
      </FormGroup>
    </FieldInterface>
  </MockFieldProvider>
</SwitchField>
`;

exports[`FormFields <SwitchField /> should render Switch with onText (custom prop) correctly 1`] = `
<SwitchField
  FieldProvider={[Function]}
  dataType="someDataType"
  id="someIdKey"
  input={
    Object {
      "name": "Name of the field",
      "value": "",
    }
  }
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
  onText="OnText"
>
  <MockFieldProvider
    dataType="someDataType"
    id="someIdKey"
    input={
      Object {
        "name": "Name of the field",
        "value": "",
      }
    }
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    onText="OnText"
    render={[Function]}
  >
    <FieldInterface
      componentType="switch-field"
      dataType="someDataType"
      id="someIdKey"
      input={
        Object {
          "name": "Name of the field",
          "onChange": [MockFunction],
          "value": "",
        }
      }
      meta={
        Object {
          "error": false,
          "touched": false,
        }
      }
      name="Name of the field"
      onText="OnText"
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <Col
            bsClass="col"
            componentClass="div"
            md={12}
          >
            <div
              className="col-md-12"
            >
              <Switch
                checked=""
                id="someIdKey"
                name="Name of the field"
                offText="OFF"
                onChange={[Function]}
                onText="OnText"
                value=""
              >
                <label
                  className="pf3-switch"
                  style={
                    Object {
                      "width": 80,
                    }
                  }
                >
                  <input
                    checked=""
                    id="someIdKey"
                    name="Name of the field"
                    onChange={[Function]}
                    type="checkbox"
                    value=""
                  />
                  <span
                    className="pf3-switch-slider"
                  >
                    <span
                      className="on-text"
                      style={
                        Object {
                          "left": -46,
                          "width": 46,
                        }
                      }
                    >
                      OnText
                    </span>
                    <span
                      className="divider"
                      style={Object {}}
                    />
                    <span
                      className="off-text"
                      style={
                        Object {
                          "left": 34,
                          "width": 46,
                        }
                      }
                    >
                      OFF
                    </span>
                  </span>
                </label>
              </Switch>
            </div>
          </Col>
        </div>
      </FormGroup>
    </FieldInterface>
  </MockFieldProvider>
</SwitchField>
`;

exports[`FormFields <SwitchField /> should render Switch with placeholder correctly 1`] = `
<SwitchField
  FieldProvider={[Function]}
  dataType="someDataType"
  id="someIdKey"
  input={
    Object {
      "name": "Name of the field",
      "value": "",
    }
  }
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
  placeholder="Placeholder"
>
  <MockFieldProvider
    dataType="someDataType"
    id="someIdKey"
    input={
      Object {
        "name": "Name of the field",
        "value": "",
      }
    }
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    placeholder="Placeholder"
    render={[Function]}
  >
    <FieldInterface
      componentType="switch-field"
      dataType="someDataType"
      id="someIdKey"
      input={
        Object {
          "name": "Name of the field",
          "onChange": [MockFunction],
          "value": "",
        }
      }
      meta={
        Object {
          "error": false,
          "touched": false,
        }
      }
      name="Name of the field"
      placeholder="Placeholder"
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <Col
            bsClass="col"
            componentClass="div"
            md={12}
          >
            <div
              className="col-md-12"
            >
              <Switch
                checked=""
                id="someIdKey"
                name="Name of the field"
                offText="OFF"
                onChange={[Function]}
                onText="ON"
                value=""
              >
                <label
                  className="pf3-switch"
                  style={
                    Object {
                      "width": 80,
                    }
                  }
                >
                  <input
                    checked=""
                    id="someIdKey"
                    name="Name of the field"
                    onChange={[Function]}
                    type="checkbox"
                    value=""
                  />
                  <span
                    className="pf3-switch-slider"
                  >
                    <span
                      className="on-text"
                      style={
                        Object {
                          "left": -46,
                          "width": 46,
                        }
                      }
                    >
                      ON
                    </span>
                    <span
                      className="divider"
                      style={Object {}}
                    />
                    <span
                      className="off-text"
                      style={
                        Object {
                          "left": 34,
                          "width": 46,
                        }
                      }
                    >
                      OFF
                    </span>
                  </span>
                </label>
              </Switch>
            </div>
          </Col>
        </div>
      </FormGroup>
    </FieldInterface>
  </MockFieldProvider>
</SwitchField>
`;

exports[`FormFields <SwitchField /> should render disabled Switch correctly 1`] = `
<SwitchField
  FieldProvider={[Function]}
  dataType="someDataType"
  id="someIdKey"
  input={
    Object {
      "name": "Name of the field",
      "value": "",
    }
  }
  isDisabled={true}
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
>
  <MockFieldProvider
    dataType="someDataType"
    id="someIdKey"
    input={
      Object {
        "name": "Name of the field",
        "value": "",
      }
    }
    isDisabled={true}
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    render={[Function]}
  >
    <FieldInterface
      componentType="switch-field"
      dataType="someDataType"
      id="someIdKey"
      input={
        Object {
          "name": "Name of the field",
          "onChange": [MockFunction],
          "value": "",
        }
      }
      isDisabled={true}
      meta={
        Object {
          "error": false,
          "touched": false,
        }
      }
      name="Name of the field"
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <Col
            bsClass="col"
            componentClass="div"
            md={12}
          >
            <div
              className="col-md-12"
            >
              <Switch
                checked=""
                disabled={true}
                id="someIdKey"
                name="Name of the field"
                offText="OFF"
                onChange={[Function]}
                onText="ON"
                value=""
              >
                <label
                  className="pf3-switch disabled"
                  style={
                    Object {
                      "width": 80,
                    }
                  }
                >
                  <input
                    checked=""
                    disabled={true}
                    id="someIdKey"
                    name="Name of the field"
                    onChange={[Function]}
                    type="checkbox"
                    value=""
                  />
                  <span
                    className="pf3-switch-slider"
                  >
                    <span
                      className="on-text"
                      style={
                        Object {
                          "left": -46,
                          "width": 46,
                        }
                      }
                    >
                      ON
                    </span>
                    <span
                      className="divider"
                      style={Object {}}
                    />
                    <span
                      className="off-text"
                      style={
                        Object {
                          "left": 34,
                          "width": 46,
                        }
                      }
                    >
                      OFF
                    </span>
                  </span>
                </label>
              </Switch>
            </div>
          </Col>
        </div>
      </FormGroup>
    </FieldInterface>
  </MockFieldProvider>
</SwitchField>
`;

exports[`FormFields <SwitchField /> should render mini Switch correctly 1`] = `
<SwitchField
  FieldProvider={[Function]}
  bsSize="mini"
  dataType="someDataType"
  id="someIdKey"
  input={
    Object {
      "name": "Name of the field",
      "value": "",
    }
  }
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
>
  <MockFieldProvider
    bsSize="mini"
    dataType="someDataType"
    id="someIdKey"
    input={
      Object {
        "name": "Name of the field",
        "value": "",
      }
    }
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    render={[Function]}
  >
    <FieldInterface
      bsSize="mini"
      componentType="switch-field"
      dataType="someDataType"
      id="someIdKey"
      input={
        Object {
          "name": "Name of the field",
          "onChange": [MockFunction],
          "value": "",
        }
      }
      meta={
        Object {
          "error": false,
          "touched": false,
        }
      }
      name="Name of the field"
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <Col
            bsClass="col"
            componentClass="div"
            md={12}
          >
            <div
              className="col-md-12"
            >
              <Switch
                bsSize="mini"
                checked=""
                id="someIdKey"
                name="Name of the field"
                offText="OFF"
                onChange={[Function]}
                onText="ON"
                value=""
              >
                <label
                  className="pf3-switch mini"
                  style={
                    Object {
                      "width": 80,
                    }
                  }
                >
                  <input
                    checked=""
                    id="someIdKey"
                    name="Name of the field"
                    onChange={[Function]}
                    type="checkbox"
                    value=""
                  />
                  <span
                    className="pf3-switch-slider"
                  >
                    <span
                      className="on-text"
                      style={
                        Object {
                          "left": -46,
                          "width": 46,
                        }
                      }
                    >
                      ON
                    </span>
                    <span
                      className="divider"
                      style={Object {}}
                    />
                    <span
                      className="off-text"
                      style={
                        Object {
                          "left": 34,
                          "width": 46,
                        }
                      }
                    >
                      OFF
                    </span>
                  </span>
                </label>
              </Switch>
            </div>
          </Col>
        </div>
      </FormGroup>
    </FieldInterface>
  </MockFieldProvider>
</SwitchField>
`;

exports[`FormFields <SwitchField /> should render readOnly Switch correctly 1`] = `
<SwitchField
  FieldProvider={[Function]}
  dataType="someDataType"
  id="someIdKey"
  input={
    Object {
      "name": "Name of the field",
      "value": "",
    }
  }
  isReadOnly={true}
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
>
  <MockFieldProvider
    dataType="someDataType"
    id="someIdKey"
    input={
      Object {
        "name": "Name of the field",
        "value": "",
      }
    }
    isReadOnly={true}
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    render={[Function]}
  >
    <FieldInterface
      componentType="switch-field"
      dataType="someDataType"
      id="someIdKey"
      input={
        Object {
          "name": "Name of the field",
          "onChange": [MockFunction],
          "value": "",
        }
      }
      isReadOnly={true}
      meta={
        Object {
          "error": false,
          "touched": false,
        }
      }
      name="Name of the field"
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <Col
            bsClass="col"
            componentClass="div"
            md={12}
          >
            <div
              className="col-md-12"
            >
              <Switch
                checked=""
                id="someIdKey"
                isReadOnly={true}
                name="Name of the field"
                offText="OFF"
                onChange={[Function]}
                onText="ON"
                value=""
              >
                <label
                  className="pf3-switch disabled"
                  style={
                    Object {
                      "width": 80,
                    }
                  }
                >
                  <input
                    checked=""
                    disabled={true}
                    id="someIdKey"
                    name="Name of the field"
                    onChange={[Function]}
                    type="checkbox"
                    value=""
                  />
                  <span
                    className="pf3-switch-slider"
                  >
                    <span
                      className="on-text"
                      style={
                        Object {
                          "left": -46,
                          "width": 46,
                        }
                      }
                    >
                      ON
                    </span>
                    <span
                      className="divider"
                      style={Object {}}
                    />
                    <span
                      className="off-text"
                      style={
                        Object {
                          "left": 34,
                          "width": 46,
                        }
                      }
                    >
                      OFF
                    </span>
                  </span>
                </label>
              </Switch>
            </div>
          </Col>
        </div>
      </FormGroup>
    </FieldInterface>
  </MockFieldProvider>
</SwitchField>
`;

exports[`FormFields <SwitchField /> should render sm Switch correctly 1`] = `
<SwitchField
  FieldProvider={[Function]}
  bsSize="mn"
  dataType="someDataType"
  id="someIdKey"
  input={
    Object {
      "name": "Name of the field",
      "value": "",
    }
  }
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
>
  <MockFieldProvider
    bsSize="mn"
    dataType="someDataType"
    id="someIdKey"
    input={
      Object {
        "name": "Name of the field",
        "value": "",
      }
    }
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    render={[Function]}
  >
    <FieldInterface
      bsSize="mn"
      componentType="switch-field"
      dataType="someDataType"
      id="someIdKey"
      input={
        Object {
          "name": "Name of the field",
          "onChange": [MockFunction],
          "value": "",
        }
      }
      meta={
        Object {
          "error": false,
          "touched": false,
        }
      }
      name="Name of the field"
    >
      <FormGroup
        bsClass="form-group"
        validationState={null}
      >
        <div
          className="form-group"
        >
          <Col
            bsClass="col"
            componentClass="div"
            md={12}
          >
            <div
              className="col-md-12"
            >
              <Switch
                bsSize="mn"
                checked=""
                id="someIdKey"
                name="Name of the field"
                offText="OFF"
                onChange={[Function]}
                onText="ON"
                value=""
              >
                <label
                  className="pf3-switch mini"
                  style={
                    Object {
                      "width": 80,
                    }
                  }
                >
                  <input
                    checked=""
                    id="someIdKey"
                    name="Name of the field"
                    onChange={[Function]}
                    type="checkbox"
                    value=""
                  />
                  <span
                    className="pf3-switch-slider"
                  >
                    <span
                      className="on-text"
                      style={
                        Object {
                          "left": -46,
                          "width": 46,
                        }
                      }
                    >
                      ON
                    </span>
                    <span
                      className="divider"
                      style={Object {}}
                    />
                    <span
                      className="off-text"
                      style={
                        Object {
                          "left": 34,
                          "width": 46,
                        }
                      }
                    >
                      OFF
                    </span>
                  </span>
                </label>
              </Switch>
            </div>
          </Col>
        </div>
      </FormGroup>
    </FieldInterface>
  </MockFieldProvider>
</SwitchField>
`;

exports[`FormFields <TextField /> should render correctly 1`] = `
<TextField
  input={
    Object {
      "name": "text-field",
    }
  }
  meta={Object {}}
>
  <FieldInterface
    componentType="text-field"
    input={
      Object {
        "name": "text-field",
      }
    }
    meta={Object {}}
    name="text-field"
  >
    <FormGroup
      bsClass="form-group"
      validationState={null}
    >
      <div
        className="form-group"
      >
        <Col
          bsClass="col"
          componentClass="div"
          md={12}
        >
          <div
            className="col-md-12"
          >
            <FormControl
              bsClass="form-control"
              componentClass="input"
              id="text-field"
              name="text-field"
            >
              <input
                className="form-control"
                id="text-field"
                name="text-field"
              />
            </FormControl>
          </div>
        </Col>
      </div>
    </FormGroup>
  </FieldInterface>
</TextField>
`;

exports[`FormFields <TextField /> should render correctly with placeholder 1`] = `
<TextField
  input={
    Object {
      "name": "text-field",
    }
  }
  meta={Object {}}
  placeholder="placeholder"
>
  <FieldInterface
    componentType="text-field"
    input={
      Object {
        "name": "text-field",
      }
    }
    meta={Object {}}
    name="text-field"
    placeholder="placeholder"
  >
    <FormGroup
      bsClass="form-group"
      validationState={null}
    >
      <div
        className="form-group"
      >
        <Col
          bsClass="col"
          componentClass="div"
          md={12}
        >
          <div
            className="col-md-12"
          >
            <FormControl
              bsClass="form-control"
              componentClass="input"
              id="text-field"
              name="text-field"
              placeholder="placeholder"
            >
              <input
                className="form-control"
                id="text-field"
                name="text-field"
                placeholder="placeholder"
              />
            </FormControl>
          </div>
        </Col>
      </div>
    </FormGroup>
  </FieldInterface>
</TextField>
`;

exports[`FormFields <TextareaField /> should render correctly 1`] = `
<TextareaField
  input={
    Object {
      "name": "textarea-field",
    }
  }
  meta={Object {}}
>
  <FieldInterface
    componentType="textarea-field"
    input={
      Object {
        "name": "textarea-field",
      }
    }
    meta={Object {}}
    name="textarea-field"
  >
    <FormGroup
      bsClass="form-group"
      validationState={null}
    >
      <div
        className="form-group"
      >
        <Col
          bsClass="col"
          componentClass="div"
          md={12}
        >
          <div
            className="col-md-12"
          >
            <FormControl
              bsClass="form-control"
              componentClass="textarea"
              id="textarea-field"
              name="textarea-field"
            >
              <textarea
                className="form-control"
                id="textarea-field"
                name="textarea-field"
              />
            </FormControl>
          </div>
        </Col>
      </div>
    </FormGroup>
  </FieldInterface>
</TextareaField>
`;

exports[`FormFields <TextareaField /> should render correctly with placeholder 1`] = `
<TextareaField
  input={
    Object {
      "name": "textarea-field",
    }
  }
  meta={Object {}}
  placeholder="placeholder"
>
  <FieldInterface
    componentType="textarea-field"
    input={
      Object {
        "name": "textarea-field",
      }
    }
    meta={Object {}}
    name="textarea-field"
    placeholder="placeholder"
  >
    <FormGroup
      bsClass="form-group"
      validationState={null}
    >
      <div
        className="form-group"
      >
        <Col
          bsClass="col"
          componentClass="div"
          md={12}
        >
          <div
            className="col-md-12"
          >
            <FormControl
              bsClass="form-control"
              componentClass="textarea"
              id="textarea-field"
              name="textarea-field"
              placeholder="placeholder"
            >
              <textarea
                className="form-control"
                id="textarea-field"
                name="textarea-field"
                placeholder="placeholder"
              />
            </FormControl>
          </div>
        </Col>
      </div>
    </FormGroup>
  </FieldInterface>
</TextareaField>
`;
